hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/BaseFlinkCommitActionExecutor.java

```diff
@@ -190,19 +191,13 @@ public Iterator<List<WriteStatus>> handleUpdate(String partitionPath, String fil
                                                   Iterator<HoodieRecord<T>> recordItr)
       throws IOException {
     // This is needed since sometimes some buckets are never picked in getPartition() and end up with 0 records
-    HoodieMergeHandle<?, ?, ?, ?> upsertHandle = (HoodieMergeHandle<?, ?, ?, ?>) this.writeHandle;
+    HoodieWriteMergeHandle<?, ?, ?, ?> upsertHandle = (HoodieWriteMergeHandle<?, ?, ?, ?>) this.writeHandle;
```

Comment:
cc @cshuo , why we need this check in like 195: `upsertHandle.isEmptyNewRecords()` ?

Reply:
We need to revise this part so that the merge handle can be configurable.

Reply:
Jira created HUDI-9618